src/language-usability.rst

```diff
@@ -16,6 +16,193 @@ Usability of Type Systems
 API Usability
 =============
 
+..
+  Outline
+  * APIs are ubiquitous
+  * Examples
+  * High level goals of API
+  * Difficulty in achieving goals of API
+  * Distinction with language usability
+..
+
+Application Programming Interfaces (APIs) are interfaces that mediate interactions between software components.
+Examples include software libraries and frameworks, OS system calls, and web protocols.
+Nearly every program written will use an API :cite:`myers2016improving`.
+
+The goal of the API designer is to create an API that a broad range of
+programmers can use in their applications.
+This goal mirrors the goals of programming language designers --
+an API designer must a create a service that is "usable" to
+a broad and vaguely scoped audience.
+Like languages, APIs must be simple enough to be used
+by users who are unaware of the inner workings of the API.
+
+In many cases, both language and API design can be approached through a "bottom-up" manner -- designed with smaller components
+that can be combined to create complex behavior.
+However, in other cases, designing APIs involves involves putting a more usable layer atop a more complex existing API.
+This adds another layer of complexity in designing APIs.
+To illustrate this diffulty:
```

Comment:
Change to "to illustrate this difficulty, Myers et al [1] write: "

Reply:
I think maybe this quote would be better in a new paragraph explaining that individual APIs are often organized into frameworks, some of which are very large: